### PR TITLE
Replace `renderToString` to `renderToStringAsync`

### PR DESCRIPTION
**Pull request by takurinton**
ref https://github.com/preactjs/wmr/issues/965#issuecomment-2005176575

## What is I did

- Update `preact-render-to-string` v6.4.0
  - Because renderToStringAsync was exported from this version
- Replace `renderToString` to `renderToStringAsync`

### CODE REVIEW

File: src/prerender.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ export default async function prerender(vnode, options) {
 	const render = () => {
 		if (++tries > maxDepth) return;
 		try {
-			return renderToString(vnode);
+			return renderToStringAsync(vnode);
 		} catch (e) {
 			if (e && e.then) return e.then(render);
 			throw e;
```
Comment (rschristian) [inline src/prerender.js:37]: _I think_ this whole block can be replaced? `renderToStringAsync` should handle that itself.

File: package.json [hunk 2]
```diff
@@ -52,6 +52,6 @@
 		"jest-environment-jsdom": "^29.7.0",
 		"jsdom": "^22.1.0",
 		"preact": "10.15.1",
-		"preact-render-to-string": "^6.2.2"
+		"preact-render-to-string": "^6.4.0"
```
Comment (rschristian) [inline package.json:55]: You'll need to bump this in `peerDependencies` too, as users won't be able to use this with the existing range of `>=5`

Comment (takurinton) [reply package.json:55]: Fixed it.
https://github.com/preactjs/preact-iso/pull/10/commits/ea1691968cea1bd90cca4909d701e9360652654b

Could this be a substantive breaking change?

Comment (rschristian) [reply package.json:55]: Eh, I'd say no.

Some people will need to upgrade, but that's not too big of an issue. I'll put it behind a new minor at least.

File: src/prerender.js (hunk 1 above)
Comment (takurinton) [reply src/prerender.js:37]: Is this your assumption?

```js
const render = () => {
	if (++tries > maxDepth) return;
	return renderToStringAsync(vnode);
};
```

Comment (rschristian) [reply src/prerender.js:37]: Can get rid of `render` altogether -- `renderToStringAsync` tracks depth and whatnot itself:

https://github.com/preactjs/preact-render-to-string/blob/e3326e2ec92e52032d8380fdc6ca5aed70fbcc48/src/index.js#L115-L128

Comment (takurinton) [reply src/prerender.js]: I see, thanks.
Fixed here
https://github.com/preactjs/preact-iso/pull/10/commits/bb3ddb0210d8f524c1e165aaf08cfc238a5a22b4

**Review by rschristian — APPROVED** [review]
Sorry, one last tiny thing: the JSDoc & types need to be corrected now that `options.maxDepth` has no effect

https://github.com/preactjs/preact-iso/blob/1a25d79e4d040d98dc12958edf5bb550e6845497/src/prerender.d.ts#L4

https://github.com/preactjs/preact-iso/blob/1a25d79e4d040d98dc12958edf5bb550e6845497/src/prerender.js#L15

Otherwise, looks good to me. Thank you so much for fixing this!

**Comment by takurinton** [comment]
@rschristian 
Thank you!! fixed!! 🙇

**Comment by rschristian** [comment]
Cheers, I'll probably wait a couple days before making a new release, just to see if anything else can get released at the same time.

**Comment by rschristian** [comment]
Just did, thanks for the reminder!

**Pull request merged**